catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -466,25 +481,31 @@ public void init() {
     get(
         "/workspaces",
         (req, res) -> {
+          Map<String, Result> workspaceMetacards;
           String email = getSubjectEmail();
           List<String> subjectRoles = getSubjectRoles();
-          Map<String, Collection<String>> attributeMap = new HashMap<>();
-          if (email != null) {
-            attributeMap.put(Core.METACARD_OWNER, Collections.singletonList(email));
-            attributeMap.put(ACCESS_ADMINISTRATORS, Collections.singletonList(email));
-            attributeMap.put(ACCESS_INDIVIDUALS, Collections.singletonList(email));
-            attributeMap.put(ACCESS_INDIVIDUALS_READ, Collections.singletonList(email));
-          }
 
-          if (CollectionUtils.isNotEmpty(subjectRoles)) {
-            attributeMap.put(ACCESS_GROUPS_READ, subjectRoles);
-            attributeMap.put(ACCESS_GROUPS, subjectRoles);
-          }
+          // TODO: DDF-4249 to refactor this logic for PreQueryPlugin Access Control
+          if (isElevatedUser(subjectRoles)) {
```

Comment:
So... @stustison are you saying we should just be doing Subject.isPermitted here and not have any isElevatedUser function?

Reply:
Yes, that is correct.